src/PhpSpec/CodeGenerator/Generator/MethodGenerator.php

```diff
@@ -45,17 +46,29 @@
     private $codeWriter;
 
     /**
-     * @param ConsoleIO $io
+     * @var StringBuilder
+     */
+    private $argumentBuilder;
+
+    /**
+     * @param ConsoleIO        $io
      * @param TemplateRenderer $templates
-     * @param Filesystem $filesystem
-     * @param CodeWriter $codeWriter
+     * @param Filesystem       $filesystem
+     * @param CodeWriter       $codeWriter
+     * @param StringBuilder    $argumentBuilder
      */
-    public function __construct(ConsoleIO $io, TemplateRenderer $templates, Filesystem $filesystem, CodeWriter $codeWriter)
-    {
-        $this->io         = $io;
-        $this->templates  = $templates;
+    public function __construct(
+        ConsoleIO $io,
+        TemplateRenderer $templates,
+        Filesystem $filesystem,
+        CodeWriter $codeWriter,
+        StringBuilder $argumentBuilder
+    ) {
+        $this->io = $io;
+        $this->templates = $templates;
         $this->filesystem = $filesystem;
         $this->codeWriter = $codeWriter;
+        $this->argumentBuilder = $argumentBuilder;
```

Comment:
The fact that you've called it `argumentBuilder` makes me think `StringBuilder` is too generic (see other comment about using `*Builder`) - reconsider this class's name

Reply:
I've called it `argumentBuilder` to reflect it's a builder for arguments, (`PhpSpec\CodeGenerator\Generator\Argument\StringBuilder` makes more sense when you read the FQCN) - I'll change the builder suffix.